docs/core/deploying/trimming-options.md

```diff
@@ -0,0 +1,125 @@
+
+## Trimming granularity
+
+The following granularity settings control how aggressively unused IL is discarded. This can be set as a property, or as metadata on an [individual assembly](#Trimmed-assemblies).
+
+- `<TrimMode>copyused</TrimMode>`
+
+   Enable assembly-level trimming, which will keep an entire assembly if any part of it is used (in a statically-understood way).
+
+- `<TrimMode>link</TrimMode>`
+
+    Enable member-level trimming, which removes unused members from types.
+
+Assemblies with `<IsTrimmable>true</IsTrimmable>` metadata but no explicit `TrimMode` will use the global `TrimMode`. The default `TrimMode` for `Microsoft.NET.Sdk` is `copyused`.
+
+## Trimmed assemblies
+
+When publishing a trimmed app, the SDK computes an `ItemGroup` called `ManagedAssemblyToLink` that represents the set of files to be processed for trimming. `ManagedAssemblyToLink` may have metadata that controls the trimming behavior per assembly. To set this metadata, create a target that runs before the built-in `PrepareForILLink` target. This example shows how to enable trimming of `MyAssembly`:
+
+```xml
+<Target Name="ConfigureTrimming"
+        BeforeTargets="PrepareForILLink">
+  <ItemGroup>
+    <ManagedAssemblyToLink Condition="'%(Filename)' == 'MyAssembly'">
+      <IsTrimmable>true</IsTrimmable>
+    </ManagedAssemblyToLink>
+  </ItemGroup>
+</Target>
+```
+
+Do not add or remove items to/from `ManagedAssemblyToLink`, because the SDK computes this set during publish and expects it not to change. The supported metadata is:
+
+- `<IsTrimmable>true</IsTrimmable>`
+
+  Control whether the given assembly is trimmed.
+
+- `<TrimMode>copyused</TrimMode>` or `<TrimMode>link</TrimMode>`
+
+  Control the [trimming granularity](#Trimming-granularity) of this assembly. This takes precedence over the global `TrimMode`. Setting `TrimMode` on an assembly implies `<IsTrimmable>true</IsTrimmable>`.
+
+## Root assemblies
+
+All assemblies which do not have `<IsTrimmable>true</IsTrimmable>` are considered roots for the analysis, which means that they and all of their statically understood dependencies will be kept. Additional assemblies may be "rooted" by name (without the `.dll` extension):
+
+```xml
+<ItemGroup>
+  <TrimmerRootAssembly Include="MyAssembly" />
+</ItemGroup>
+```
+
+## Root descriptors
+
+Another way to specify roots for analysis is using an XML file that uses the linker [descriptor format](https://github.com/mono/linker/blob/master/docs/data-formats.md#descriptor-format). This lets you root specific members instead of a whole assembly.
+
+```xml
+<ItemGroup>
+  <TrimmerRootDescriptor Include="MyRoots.xml" />
+</ItemGroup>
+```
+
+For example, `MyRoots.xml` might root a specific method that is dynamically accessed by the application:
+
+```xml
+<linker>
+  <assembly fullname="MyAssembly">
+    <type fullname="MyAssembly.MyClass">
+      <method signature="System.Void DynamicallyAccessedMethod()" />
```

Comment:
Let's use the `name="DynamicallyAccessedMethod"` instead of the signature - it's easier - getting the signature right for more complex methods can be rather tricky.